### PR TITLE
COL-669 Include large image in YouTube preview generation

### PR DESCRIPTION
**Pull request by paulkerschen**
Also includes improvements to small thumbnail generation. See https://jira.ets.berkeley.edu/jira/browse/COL-669 for details.

### CODE REVIEW

File: lib/preview/link.js [hunk 1]
```diff
@@ -102,18 +102,41 @@ var getYoutubeId = function(link) {
...
-      'youtubeId': youtubeId
-    };
-    var result = new Result(PreviewConstants.STATUS.DONE, thumbnailPath, null, null, metadata);
-    return callback(null, result);
+    PreviewImage.generateThumbnail(ctx, thumbnailDownloadPath, function(err, thumbnailPath) {
+      if (err) {
+        return callback(err);
+      }
+
+      var previewUrl = util.format('http://img.youtube.com/vi/%s/maxresdefault.jpg', youtubeId);
```
Comment (johncrossman) [inline lib/preview/link.js]: @paulkerschen will this "max res" preview URL work in call cases? I do not get our beloved Winona + pizza with https://img.youtube.com/vi/mzDwKMgsfmo/maxresdefault.jpg BUT the mqdefault URL works: https://img.youtube.com/vi/mzDwKMgsfmo/mqdefault.jpg  

Expected behavior:
image

Comment (paulkerschen) [inline lib/preview/link.js]: You're right, @johncrossman - seems like hqdefault.jpg is the largest image that's guaranteed to be there. Amended and testing on dev; will signal once it's checked out.

**Comment by paulkerschen** [comment]
Thanks for sanity check @johncrossman; amended and tested on suitec-preview-service-dev.

**Comment by johncrossman** [comment]
Winona thanks you, @paulkerschen

**Pull request merged**